tg_bot.py

```diff
@@ -108,9 +119,113 @@ def quering_termins(update, context, reuse=False):
     if not found_any:
         msg.reply_text('Unfortunately, everything is booked. Please come back in several days :(')
 
+    buttons = [InlineKeyboardButton(text="Schedule", callback_data="schedule"),
+               InlineKeyboardButton(text="No, return back", callback_data="return")]
+    custom_keyboard = [buttons]
+
+    msg.reply_text(
+        'If you want, you can schedule checking appointments of this type by interval',
+        reply_markup=InlineKeyboardMarkup(custom_keyboard, one_time_keyboard=True))
+
+    return SCHEDULE_APPOINTMENT
+
+
+def set_retry_interval(update, context):
+    if update.callback_query and update.callback_query.data == 'return':
+        return selecting_buro(update, context)
+    else:
+        msg = update.callback_query.message if update.callback_query else update.message
+        msg.reply_text('Please type interval in minutes. Interval should greater or equals 15 minutes.')
+        return SELECT_INTERVAL
+
+
+def print_available_termins(update, context):
+    """
+    Checks for available termins and prints them if any
+    """
+    department = context.user_data['buro']
+    termin_type_str = context.user_data['termin_type']
+
+    msg = update.message
+
+    appointments = termin.get_termins(department, termin_type_str)
+    found_any = False
+    for k, v in appointments.items():
+        for date in v['appoints']:
+            if v['appoints'][date]:
+                found_any = True
+                msg.reply_text('Available appointments on date %s are %s' % (date, '\n'.join(v['appoints'][date])))
```

Comment:
Please print only nearest available one. I just tried and it returned me 56 messages 1 screen long each. Do not print everything :)

Reply:
Actually, I think it should be an option - I was using this for applying for BK and available appointments on this are so rare (everything is booked for several months). Plus sometimes someone cancels his appointment right for the next day and I don't want to loose available appointments on next months because of this. In other words, I am looking for an appointment in several months and want to see all available appointments. 
What do you think?

Reply:
Maybe I misunderstood but you still looks for an appointment as soon as possible, correct? And even if you grabbed one two months in advance, you still will continue checking to try to get one sooner.

Displaying all available data is possible (since we get it from the servers anyway), I don't mind having an option, but this should not be default behavior, because I believe, usually you just want appointment as soon as possible. IMO it's not really important that there are free slots 3 month in advance if there are slots available for next week.

Reply:
It also can be "2-3 nearest days" by default instead of one day, should be also fine.

Reply:
Ok, I think we can leave printing only nearest days for now